### ISSUE TITLE
MPI_Neighbor_alltoall with empty graph

### ISSUE THREAD

**Issue by ibaned** [open-mpi/ompi#2402]
This is a Github version of https://www.mail-archive.com/devel@lists.open-mpi.org//msg19753.html

There is a regression in the use of neighborhood collectives with empty topologies.
The following program:

```
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
  MPI_Init(&argc, &argv);
  int n = 1;
  int sources[1] = {0};
  int degrees[1] = {0};
  int destinations[1];
  int reorder = 0;
  MPI_Comm graph_comm;
  MPI_Dist_graph_create(MPI_COMM_WORLD, n, sources, degrees, destinations,
      MPI_UNWEIGHTED, MPI_INFO_NULL, reorder, &graph_comm);
  int sendbuf[1];
  int recvbuf[1];
  MPI_Neighbor_alltoall(sendbuf, 1, MPI_INT,
      recvbuf, 1, MPI_INT, graph_comm);
  MPI_Comm_free(&graph_comm);
  MPI_Finalize();
}
```

produces errors like these when compiled against OpenMPI at
versions `v2.0.1` and `a49422f`, but runs okay at version `v1.10.4`:

```
[Dans-MacBook-Air:52159] *** An error occurred in MPI_Neighbor_alltoall
[Dans-MacBook-Air:52159] *** reported by process [2935488513,0]
[Dans-MacBook-Air:52159] *** on communicator MPI COMMUNICATOR 3 CREATE FROM 0
[Dans-MacBook-Air:52159] *** MPI_ERR_INTERN: internal error
[Dans-MacBook-Air:52159] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[Dans-MacBook-Air:52159] ***    and potentially your MPI job)
```

**Comment by jsquyres** [comment]
@ibaned Thanks for reporting!  I think @bosilca fixed this issue in commit 725277b.  Can you try the latest Open MPI nightly master snapshot to see if it is fixed for you?

**Comment by ibaned** [comment]
Confirmed !
Compiling at commit fb5ccd3, my test program above runs fine and so does most of my real test suite. There is a different error printed for runs over 2 ranks, but I will open a different Github issue for that one. I appreciate the fast response from your team.
